model/lifecycle.go

```diff
@@ -365,14 +365,8 @@ func RestartVersion(versionId string, taskIds []string, abortInProgress bool, ca
 		Build     string
 		TaskGroup string
 	}
-	// Mark aborted tasks to reset when finished if not all tasks are finished.
-	if abortInProgress && len(finishedTasks) < len(taskIds) {
```

Comment:
why do we no longer need this?

Reply:
Mentioned this in the comment but I modified the function at the top AbortTasksForVersion to be AbortAndMarkResetTasksForVersion. Potentially there's a reason we don't do this together already, but Aborted tasks aren't archived since we only archive finished tasks so I don't see why it needs to be done separately.